docs/specs/sections/implementation.tex

```diff
@@ -0,0 +1,89 @@
+
+\subsection{Participants} 
+Protocol implementation involves realization of the protocol building blocks as well as providing means of data communication between them. Building blocks are placed at the following locations, which correspond to protocol participants:
+
+\begin{itemize}%[label=$\bullet$]
+	\item User software.
+	\item License Provider software.
+	\item Service Provider software.
+	\item Citadel contract.
+\end{itemize}
+
+\begin{flushleft}
+Data communication between protocol participants is realized in the following communication modes:
+\end{flushleft}
+
+\begin{itemize}%[label=$\bullet$]
+	\item Transactions which change the contract state.
+	\item Queries which do not change the contract state.
+	\item Storing data directly in blockchain by issuing transactions.
+	\item Retrieving data from blockchain by scanning transactions.
+	\item Delegating ZK proof calculation.
+	\item Off-chain calls.
+\end{itemize}
```

Comment:
I am not sure I agree, they are all types of calls - but I will reformulate them so that it is clearer and more concrete. There is no semantic here, only technicalities of data transfer.